google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/STBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,211 @@
+            ITBigQueryStorageLongRunningTest.class.getSimpleName(), parentProjectId));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (client != null) {
+      client.close();
+    }
+    if (bigquery != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, DATASET);
+      LOG.info("Deleted test dataset: " + DATASET);
+    }
+  }
+
+  @Test
+  public void testDefaultStream() throws IOException, InterruptedException, ExecutionException,
+      Descriptors.DescriptorValidationException {
+    // Set up a default stream. Write to it for a long time, (a few minutes for now) and make
+    // sure that everything goes well.
+    String tableName = "JsonTableDefaultStream";
+    TableInfo tableInfo =
+        TableInfo.newBuilder(
+            TableId.of(DATASET, tableName),
+            StandardTableDefinition.of(
+                Schema.of(
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_str", StandardSQLTypeName.STRING)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_numerics", StandardSQLTypeName.NUMERIC)
+                        .setMode(Field.Mode.REPEATED)
+                        .build(),
+                    com.google.cloud.bigquery.Field.newBuilder(
+                        "test_datetime", StandardSQLTypeName.DATETIME)
+                        .build())))
+            .build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(parent.toString(), tableInfo.getDefinition().getSchema())
+            .createDefaultStream()
+            .setBatchingSettings(
+                StreamWriter.Builder.DEFAULT_BATCHING_SETTINGS
+                    .toBuilder()
+                    .setRequestByteThreshold(1024 * 1024L) // 1 Mb
+                    .setElementCountThreshold(2L)
+                    .setDelayThreshold(Duration.ofSeconds(2))
+                    .build())
+            .build()) {
+      for (int i = 0; i < 5; i++){
+        LOG.info("Sending a message");
+        // Ramping up the size increases the latency
+        JSONObject row = MakeJsonObject(1);
+        JSONArray jsonArr = new JSONArray(new JSONObject[] {row});
+
+        LocalDateTime start = LocalDateTime.now();
+        Date startTime = new Date();
+        //TODO(jstocklass): Make asynchronized calls instead of synchronized calls
+        ApiFuture<AppendRowsResponse> response = jsonStreamWriter.append(jsonArr, -1);
+        assertEquals(0, response.get().getAppendResult().getOffset().getValue());
+        Assert.assertFalse(response.get().getAppendResult().hasOffset());
+        Date finishTime = new Date();
+        LOG.info("Latency: ".concat(String.valueOf(finishTime.getTime() - startTime.getTime())).concat(" ms"));
```

Comment:
Can you add a TODO to compute aggregate statistics (quantiles, histograms, etc) instead of logging this to debug logs? Once we have a large number of requests, the debug logs will be completely filled with this.

Also in Java you can use + to concatenate strings, which is cleaner than calling concat().

Reply:
Done